napalm_eos/eos.py

```diff
@@ -1201,3 +1201,143 @@ def traceroute(self, destination, source='', ttl=0, timeout=0):
+                if int(peer_info['remote_as']) not in bgp_detail_info:
+                    bgp_detail_info[int(peer_info['remote_as'])] = []
+
+                bgp_detail_info[int(peer_info['remote_as'])].append(peer_info)
+
+        return bgp_detail_info
+
+    def _parse_per_peer_bgp_detail(self, peer_output):
+        """This function parses the raw data per peer and returns a
+        json structure per peer.
+        """
+
+        int_fields = ['accepted_prefix_count', 'local_as', 'remote_as',
+                      'local_port', 'remote_port', 'local_port',
+                      'input_messages', 'output_messages', 'input_updates',
+                      'output_updates', 'messages_queued_out', 'holdtime',
+                      'configured_holdtime', 'keepalive',
+                      'configured_keepalive', 'advertise_prefix_count',
+                      'received_prefix_count']
+
+        peer_details = []
+
+        # Using preset template to extract peer info
+        peer_info = (
+            napalm_base.helpers.textfsm_extractor(
+                self, 'bgp_detail', peer_output))
+
+        for item in peer_info:
+            # Issuing command to get number of accepted prefixes for peer
+            try:
+                peer_ver = IPAddress(item['remote_address']).version
+            except Exception:
+                continue
+
+            if peer_ver == 4:
+                acc_prefix_command = (
+                    'show ip bgp summary | include %s' %
+                    item['remote_address'])
+            elif peer_ver == 6:
+                acc_prefix_command = (
+                    'show ipv6 bgp summary | include %s' %
+                    item['remote_address'])
+
+            accepted_prefix_output = (
+                self.device.run_commands([acc_prefix_command],
+                                         encoding='text')[0]['output'])
+
+            accepted_prefix_count = accepted_prefix_output.split()[-1]
+
+            # Determining a few other fields in the final peer_info
+            item['up'] = (
+                True if item['up'] == "up" else False)
+            item['local_address_configured'] = (
+                True if item['local_address'] else False)
+            item['multihop'] = (
+                False if item['multihop'] == 0 or
+                item['multihop'] == '' else True)
+
+            # TODO: The below fields need to be retrieved
+            # Currently defaulting their values to False or 0
+            item['multipath'] = False
+            item['remove_private_as'] = False
+            item['suppress_4byte_as'] = False
+            item['local_as_prepend'] = False
+            item['flap_count'] = 0
+
+            # Converting certain fields into int
+            for key in int_fields:
+                try:
+                    item[key] = (
+                        napalm_base.helpers.convert(int, item[key], 0))
+                except Exception:
```

Comment:
You don't need to catch any exception here, that's the advantage of using this helper.